playback/src/player.rs

```diff
@@ -43,6 +44,52 @@ enum PlayerCommand {
     Seek(u32),
 }
 
+#[derive(Clone, Copy, Debug)]
+struct NormalisationData {
+    track_gain_db: f32,
+    track_peak: f32,
+    album_gain_db: f32,
+    album_peak: f32,
+}
+
+impl NormalisationData {
+    fn new<T: Read + Seek>(file: &mut AudioDecrypt<T>) -> NormalisationData {
+        static SPOTIFY_HEADER_START_OFFSET: u64 = 144;
+        file.seek(SeekFrom::Start(SPOTIFY_HEADER_START_OFFSET)).unwrap();
+
+        let track_gain_db: f32 = file.read_f32::<LittleEndian>().unwrap();
+        let track_peak: f32 = file.read_f32::<LittleEndian>().unwrap();
+        let album_gain_db: f32 = file.read_f32::<LittleEndian>().unwrap();
+        let album_peak: f32 = file.read_f32::<LittleEndian>().unwrap();
+
+        NormalisationData {
+            track_gain_db: track_gain_db,
+            track_peak: track_peak,
+            album_gain_db: album_gain_db,
+            album_peak: album_peak,
+        }
+    }
+
+    fn get_factor(config: &PlayerConfig, data: NormalisationData) -> f32 {
```

Comment:
Since you already have a `if self.config.normalisation && normalisation_factor != 1.0` condition before applying it, you can ignore the config here.

Reply:
missed that. removed.